curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/655b49333d9f265bc1512152.md

```diff
@@ -27,10 +27,15 @@ You should create a `currentSongIndex` constant and set it to the calling of the
 assert.match(code, /audio\.addEventListener\(('|")ended\1,\s*\(\)\s*=>\s*\{\s*const\s+currentSongIndex\s*=\s*getCurrentSongIndex\(\);?\s*.*\s*\}\);?/)
 ```
 
-You should check if a next song exists with `userData?.songs[currentSongIndex + 1] !== undefined` and set it to a `nextSongExists` constant.
+You should check if a next song exists comparing `userData.songs.length` and `currentSongIndex` and set it to a `nextSongExists` constant. If the last index of the songs array (`userData.songs.length - 1`) is bigger than the currentSongIndex that means there is a next song.
 
 ```js
-assert.match(code, /const\s+nextSongExists\s*=\s*userData\?\.songs\s*\[\s*currentSongIndex\s*\+\s*1\s*\]\s*!==\s*undefined;?/)
+const secondPart = code.split(`audio.addEventListener("ended", () => {`)[1];
```

Comment:
we should allow for this solution to pass `currentSongIndex > userData?.songs.length - 1`

<img width="874" alt="Screenshot 2024-02-03 at 7 57 08 AM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/5109161e-8eff-46c3-84ab-6b9e36e1a0a6">

It will only pass if I change it to a ternary.
But if we are allowing for any valid answer, then that answer should pass too since it evaluates to a boolean

Reply:
But this solution is wrong, if the currentSongIndex is bigger than the last song position that means there is no next music. I believe it is possible you inverted the "<" signal.

Reply:
> But this solution is wrong, if the currentSongIndex is bigger than the last song position that means there is no next music.

Right, which means it will evulate to false here
```
currentSongIndex > userData?.songs.length - 1

```
my question is why are forcing a ternary here
```
currentSongIndex > userData?.songs.length - 1 ? false : true
```
when this returns a boolean anyway

```
currentSongIndex > userData?.songs.length - 1

```

aren't both answers the same?

Reply:
There is no test in the code that enforces ternary , the thing the blocks of code you mentioned are exactly the opposite. The first one has the assertion of the second one but the ternary inverts the boolean value. [assertion] ? [if assertion is true] return false : [if assertion is false] return true. That is why the first block of code is correct and the second isn't. It would be the same if it was [assertion] ? true : false.

Reply:
Now, I get it 👍